osbenchmark/worker_coordinator/runner.py

```diff
@@ -651,6 +651,72 @@ def error_description(self, error_details):
     def __repr__(self, *args, **kwargs):
         return "bulk-index"
 
+class TrainKNNModel(Runner): 
+    """
+    Trains model named model_id until training is complete or timeout is exceeded. 
+    """
+
+    NAME = "train-knn-model"
+    
+    async def __call__(self, opensearch, params):
+        """
+        Create and train one model named model_id. 
+        
+        :param opensearch: The OpenSearch client.
+        :param params: A hash with all parameters. See below for details.
+        :return: A hash with meta data for this bulk operation. See below for details.
+        :raises: Exception if training fails, times out, or a different error occurs. 
+        It expects a parameter dict with the following mandatory keys:
+
+        * ``body``: containing parameters to pass on to the train engine. 
+            See https://opensearch.org/docs/latest/search-plugins/knn/api/#train-a-model for information. 
+        * ``retries``: Maximum number of retries allowed for the training to complete (seconds).
+        * ``polling-interval``: Polling interval to see if the model has been trained yet (seconds). 
+        * ``model_id``: ID of the model to train. 
+        """
+        body = params["body"]
+        model_id = parse_string_parameter("model_id", params)
+        max_retries = parse_int_parameter("retries", params)
+        poll_period = parse_float_parameter("poll_period", params)
+
+        method = "POST"
+        model_uri = "/_plugins/_knn/models/{}".format(model_id)
+
+        await opensearch.transport.perform_request(method, "{}/_train".format(model_uri), body=body)
+        
+        current_number_retries = 0
+        while True: 
+            model_response = await opensearch.transport.perform_request("GET", model_uri)
+
+            if 'state' not in model_response.keys() or current_number_retries > max_retries:
```

Comment:
i don't think you can combine both the condition. Can you please break it into two condition and raise appropriate exception message?